### PR TITLE
*: support foreign key on update cascade and set null when execute update statement

### LINKED ISSUE #38651: support foreign key on update cascade and set null
## Dev Task

support foreign key on update cascade and set null

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Issue Number: close #38651

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by you06 — APPROVED** [review]

File: executor/foreign_key.go [hunk 1]
```diff
@@ -735,6 +824,8 @@ func genFKValueString(v types.Datum) (string, error) {
 	switch v.Kind() {
 	case types.KindInt64, types.KindUint64, types.KindFloat32, types.KindFloat64, types.KindMysqlDecimal:
 		return val, nil
+	case types.KindNull:
+		return "NULL", err
 	default:
 		return "'" + val + "'", nil
```
Comment (ekexium) [inline executor/foreign_key.go:830]: Do all remaining types work well with `'`?

**Review by ekexium — APPROVED** [review]
I suggest adding some comments explaning how it works. It's not so obvious at first sight.

Are there still any difficulties stopping us from building the AST instead of manipulating SQLs? I'm somewhat worried datum -> string(sql) -> datum may not be identical(?)

File: executor/foreign_key.go (hunk 1 above)
Comment (crazycs520) [reply executor/foreign_key.go]: nice catch, I just add all types test in `mysql_test/t/foreign_key_cascade_type.test` https://github.com/pingcap/tidb-test/pull/2000.

**Comment by crazycs520** [comment]
> I suggest adding some comments explaning how it works. It's not so obvious at first sight.
> 
> Are there still any difficulties stopping us from building the AST instead of manipulating SQLs? I'm somewhat worried datum -> string(sql) -> datum may not be identical(?)

I'll try this later.

**Pull request merged**